test/unit/gateways/cyber_source_rest_test.rb

```diff
@@ -233,6 +233,166 @@ def test_url_building
+      assert_nil json_body['processingInformation']['authorizationOptions']['ignoreCvResult']
+      true
+    end.returns(successful_purchase_response)
+
+    options = @options.merge(ignore_avs: 'false')
+    assert response = @gateway.purchase(@amount, @credit_card, options)
+    assert_success response
+  end
+
+  def test_successful_credit_card_purchase_single_request_ignore_ccv
+    @gateway.expects(:ssl_post).with do |_host, request_body|
+      json_body = JSON.parse(request_body)
+      assert_nil json_body['processingInformation']['authorizationOptions']['ignoreAvsResult']
+      assert_equal json_body['processingInformation']['authorizationOptions']['ignoreCvResult'], 'true'
+      true
+    end.returns(successful_purchase_response)
+
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(
+                                                                 ignore_cvv: true
+                                                               ))
+    assert_success response
+  end
+
+  def test_successful_credit_card_purchase_single_request_without_ignore_ccv
+    @gateway.expects(:ssl_post).with do |_host, request_body|
+      json_body = JSON.parse(request_body)
+      assert_nil json_body['processingInformation']['authorizationOptions']['ignoreAvsResult']
+      assert_nil json_body['processingInformation']['authorizationOptions']['ignoreCvResult']
+      true
+    end.returns(successful_purchase_response)
+
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(
+                                                                 ignore_cvv: false
+                                                               ))
+    assert_success response
+
+    @gateway.expects(:ssl_post).with do |_host, request_body|
+      json_body = JSON.parse(request_body)
+      assert_nil json_body['processingInformation']['authorizationOptions']['ignoreAvsResult']
+      assert_nil json_body['processingInformation']['authorizationOptions']['ignoreCvResult']
+      true
+    end.returns(successful_purchase_response)
+
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(
+                                                                 ignore_cvv: 'false'
+                                                               ))
+    assert_success response
+  end
+
+  def test_purchase_includes_mdd_fields
+    stub_comms do
+      @gateway.purchase(100, @credit_card, order_id: '1', mdd_field_2: 'CustomValue2', mdd_field_3: 'CustomValue3')
+    end.check_request do |_endpoint, data, _headers|
+      json_data = JSON.parse(data)
+      assert_equal json_data['merchantDefinedInformation'][0]['key'], 'mdd_field_2'
+      assert_equal json_data['merchantDefinedInformation'][0]['value'], 'CustomValue2'
+      assert_equal json_data['merchantDefinedInformation'].count, 2
+    end.respond_with(successful_purchase_response)
+  end
+
+  def test_authorize_includes_mdd_fields
```

Comment:
❓ Question:

Any particular reason to test purchase and authorize by separate?, both rely on `build_auth_request` method

Reply:
just following the practice on cybersource old version, but considering what you say, just keep the authorize. Thanks!